sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -409,9 +447,18 @@ private void initializeDatasetService(PipelineOptions pipelineOptions) {
       }
     }
 
+    @Setup
+    public void setup() {
+      if (useDefaultStream) {
+        destinations = Maps.newHashMap();
+      }
+    }
+
     @StartBundle
     public void startBundle() throws IOException {
-      destinations = Maps.newHashMap();
+      if (!useDefaultStream) {
```

Comment:
Why add this if?

Reply:
Only want to reuse the destination state for the default stream clients.